tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspClientTool/TspClientTool.cs

```diff
@@ -0,0 +1,529 @@
+            updateCommand.SetHandler(async ctx => await HandleCommand(ctx, ctx.GetCancellationToken()));
+            syncCommand.SetHandler(async ctx => await HandleCommand(ctx, ctx.GetCancellationToken()));
+            generateCommand.SetHandler(async ctx => await HandleCommand(ctx, ctx.GetCancellationToken()));
+
+            // Add subcommands to root
+            rootCommand.AddCommand(initCommand);
+            rootCommand.AddCommand(updateCommand);
+            rootCommand.AddCommand(syncCommand);
+            rootCommand.AddCommand(generateCommand);
+
+            return rootCommand;
+        }
+
+        public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            try
+            {
+                var commandResult = ctx.ParseResult.CommandResult;
+                var commandName = commandResult.Command.Name;
+
+                logger.LogInformation("Executing tsp-client command: {CommandName}", commandName);
+
+                var response = commandName switch
+                {
+                    initCommandName => await ExecuteInitCommand(ctx),
+                    updateCommandName => await ExecuteUpdateCommand(ctx),
+                    syncCommandName => await ExecuteSyncCommand(ctx),
+                    generateCommandName => await ExecuteGenerateCommand(ctx),
+                    _ => new DefaultCommandResponse { ResponseError = $"Unknown command: {commandName}" }
+                };
+
+                ctx.ExitCode = ExitCode;
+                output.Output(response);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error executing tsp-client command");
+                SetFailure(1);
+                ctx.ExitCode = ExitCode;
+                output.Output(new DefaultCommandResponse
+                {
+                    ResponseError = $"Error executing tsp-client command: {ex.Message}"
+                });
+            }
+
+            await Task.CompletedTask;
+        }
+
+        [McpServerTool(Name = "tsp-client-init"), Description("Initialize SDK project folder from tspconfig.yaml")]
+        public async Task<DefaultCommandResponse> InitializeProject(
+            string tspConfigPath,
+            string? outputDir = null,
+            bool skipSyncAndGenerate = false,
+            string? localSpecRepo = null,
+            string? commit = null,
+            string? repo = null,
+            bool skipInstall = false,
+            string? emitterOptions = null,
+            string? emitterPackageJsonPath = null,
+            bool saveInputs = false,
+            bool debug = false)
+        {
+            try
+            {
+                // Validate tspconfig path exists
+                if (!File.Exists(tspConfigPath))
+                {
+                    // Check if it's a directory containing tspconfig.yaml
+                    var tspConfigInDir = Path.Combine(tspConfigPath, "tspconfig.yaml");
```

Comment:
I believe this check already exists in tsp-client, let's not duplicate this logic